src/App.tsx

```diff
@@ -17,18 +65,22 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                filter={filter}
+                onFilterChange={setFilter}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!isLoaded
+                ? <Loader />
+                : <TodoList todos={visibleTodos} onTodoSelect={setModalTodo} selectedTodo={modalTodo} />}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {Object.keys(modalTodo).length > 0 && <TodoModal todo={modalTodo} onModalChange={setModalTodo} />}
```

Comment:
better set `null` by default instead of an empty object and then check here will look like `{modalTodo && <TodoModal propExample={yourProps} />}`